ingestion/src/metadata/ingestion/source/metadata/atlas/metadata.py

```diff
@@ -340,72 +344,77 @@ def _parse_table_columns(self, table_response, tbl_entity, name) -> List[Column]
-            ]
-            if not tbl_entity["referredEntities"].get(key):
-                continue
-            table_name = tbl_entity["referredEntities"][key]["relationshipAttributes"][
-                "table"
-            ]["displayText"]
-            from_fqn = fqn.build(
-                self.metadata,
-                entity_type=Table,
-                service_name=self.service.name.__root__,
-                database_name=db_entity["displayText"],
-                schema_name=db_entity["displayText"],
-                table_name=table_name,
-            )
-            from_entity_ref = self.get_lineage_entity_ref(
-                from_fqn, self.metadata_config, "table"
+        try:
+            lineage_response = self.atlas_client.get_lineage(source_guid)
+            lineage_relations = lineage_response["relations"]
+            tbl_entity = self.atlas_client.get_entity(
+                lineage_response["baseEntityGuid"]
             )
-            for edge in lineage_relations:
-                if (
-                    lineage_response["guidEntityMap"][edge["toEntityId"]]["typeName"]
-                    == "processor"
+            for key in tbl_entity["referredEntities"].keys():
+                if not tbl_entity["entities"][0]["relationshipAttributes"].get(
+                    self.entity_types["Table"][name]["db"]
                 ):
                     continue
-
-                tbl_entity = self.atlas_client.get_entity(edge["toEntityId"])
-                for key in tbl_entity["referredEntities"]:
-                    db_entity = tbl_entity["entities"][0]["relationshipAttributes"][
-                        self.entity_types["Table"][name]["db"]
-                    ]
-
-                    db = self.get_database_entity(db_entity["displayText"])
-                    table_name = tbl_entity["referredEntities"][key][
-                        "relationshipAttributes"
-                    ]["table"]["displayText"]
-                    to_fqn = fqn.build(
-                        self.metadata,
-                        entity_type=Table,
-                        service_name=self.service.name.__root__,
-                        database_name=db.name.__root__,
-                        schema_name=db_entity["displayText"],
-                        table_name=table_name,
-                    )
-                    to_entity_ref = self.get_lineage_entity_ref(
-                        to_fqn, self.metadata_config, "table"
-                    )
-                    yield from self.yield_lineage(from_entity_ref, to_entity_ref)
+                db_entity = tbl_entity["entities"][0]["relationshipAttributes"][
+                    self.entity_types["Table"][name]["db"]
+                ]
+                if not tbl_entity["referredEntities"].get(key):
+                    continue
+                table_name = tbl_entity["referredEntities"][key][
+                    "relationshipAttributes"
+                ]["table"]["displayText"]
```

Comment:
let's bring this in but eventually this should become a pydantic model as well since we are having a lot of dict levels